plugins/draco/io/DracoReader.cpp

```diff
@@ -0,0 +1,363 @@
+                layout->registerDim(Dimension::Id::NormalY);
+                layout->registerDim(Dimension::Id::NormalZ);
+                break;
+            }
+            case GeometryAttribute::COLOR:
+            {
+                layout->registerDim(Dimension::Id::Red);
+                layout->registerDim(Dimension::Id::Green);
+                layout->registerDim(Dimension::Id::Blue);
+            }
+            case GeometryAttribute::TEX_COORD:
+            {
+                layout->registerDim(Dimension::Id::TextureU);
+                layout->registerDim(Dimension::Id::TextureV);
+                if (nc == 3)
+                {
+                    m_textureW = true;
+                    layout->registerDim(Dimension::Id::TextureW);
+                }
+                break;
+            }
+
+            case GeometryAttribute::GENERIC:
+            {
+                Dimension::Id id = pdal::Dimension::id(name);
+                if (id != Dimension::Id::Unknown) {
+                    m_generics[id] = std::vector<double>(m_pc->num_points(), 0.0);
+                    layout->registerOrAssignDim(name, getPdalType(dt));
+                }
+                break;
+            }
+
+            default:
+                // Not supported draco domain types
+                throw pdal_error("Unknown Geometry Attribute Type");
+                break;
+        }
+    }
+}
+
+void DracoReader::ready(PointTableRef)
+{
+    using namespace draco;
+    for (int at_id=0; at_id < m_pc->num_attributes(); ++at_id)
+    {
+        const PointAttribute* attr = m_pc->GetAttributeByUniqueId(at_id);
+        draco::DataType dt = attr->data_type();
+        draco::GeometryAttribute::Type at = attr->attribute_type();
+
+        std::string name;
+        const AttributeMetadata* attr_metadata = m_pc->GetAttributeMetadataByAttributeId(at_id);
+        if (attr_metadata)
+        {
+            attr_metadata->GetEntryString("name", &name);
+        }
+
+        switch (at)
+        {
+            case GeometryAttribute::POSITION:
+                CopyAttributeData<double>(m_count, attr, m_positions);
+                break;
+            case GeometryAttribute::NORMAL:
+                CopyAttributeData<double>(m_count, attr, m_normals);
+                break;
+            case GeometryAttribute::COLOR:
+                CopyAttributeData<uint16_t>(m_count, attr, m_colors);
+            case GeometryAttribute::TEX_COORD:
+                CopyAttributeData<double>(m_count, attr, m_textures);
+                break;
+
+            case GeometryAttribute::GENERIC:
+                if (name.length() > 0)
+                {
+                    Dimension::Id id = Dimension::id(name);
+                    CopyAttributeData<double>(1, attr, m_generics.at(id));
+                }
+
+                break;
+
+            default:
+                // Unsupported draco domain types
+                throw pdal_error("Unknown Geometry Attribute Type");
```

Comment:
Tell the user more information why we failed here. What geometry type were we given that we can't read?